components/gitpod-protocol/src/protocol.ts

```diff
@@ -356,6 +356,68 @@ export namespace UserEnvVar {
     }
 }
 
+export interface SSHPublicKeyValue {
+    name: string;
+    key: string;
+}
+export interface UserSSHPublicKey extends SSHPublicKeyValue {
+    id: string;
+    key: string;
+    userId: string;
+    fingerprint: string;
+    creationTime: string;
+    lastUsedTime?: string;
+}
+
+export type UserSSHPublicKeyValue = Omit<UserSSHPublicKey, "key" | "userId">;
+
+export namespace SSHPublicKeyValue {
+    export function validate(value: SSHPublicKeyValue): string | undefined {
```

Comment:
**issue(non-blocking)**: What do you think of also improving the error message when trying to add the same key again?

| Duplicate Key Error |
|-|
| <img width="1440" alt="Screenshot 2022-06-16 at 7 06 37 PM" src="https://user-images.githubusercontent.com/120486/174116130-342d16a4-a472-4554-a2cd-8553fc926d52.png"> |